paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -1010,6 +993,34 @@ internal class CustomerSheetViewModel(
             }
     }
 
+    private suspend fun refreshAndUpdatePaymentMethods(
+        newPaymentMethod: PaymentMethod
+    ) {
+        awaitCustomerAdapter().retrievePaymentMethods().onSuccess { paymentMethods ->
+            val selection = PaymentSelection.Saved(newPaymentMethod)
+
+            safeUpdateSelectPaymentMethodState {
+                it.copy(
+                    savedPaymentMethods = sortPaymentMethods(paymentMethods, selection),
+                    paymentSelection = PaymentSelection.Saved(newPaymentMethod),
+                    primaryButtonVisible = true,
+                    primaryButtonLabel = resources.getString(
+                        R.string.stripe_paymentsheet_confirm
+                    ),
+                )
+            }
+
+            onBackPressed()
+        }.onFailure { exception, _ ->
+            errorReporter.report(
+                ErrorReporter.ExpectedErrorEvent.CUSTOMER_SHEET_PAYMENT_METHODS_REFRESH_FAILURE,
+                StripeException.create(exception),
+            )
+
+            onDismissed()
```

Comment:
`iOS` does this where they simply cancel if we are not able to refetch payment methods. We don't really have any refresh concept in `PaymentSheet` or `CustomerSheet`. I'm wondering if we should at least fail here with an error.

Reply:
Hm, yeah, reporting some kind of error here feels better than just cancelling. I think if you do something like that, we should probably make the same change on iOS? It might make sense to coordinate and then update this in a follow up.